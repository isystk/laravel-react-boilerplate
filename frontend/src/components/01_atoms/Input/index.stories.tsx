import { Meta, Story } from '@storybook/react'
import React from 'react'
import Input, { InputProps } from './index'
import { useForm } from 'react-hook-form'

export default {
  title: '01_atoms/Input',
  component: Input,
} as Meta

type FormInputs = {
  email: string
}

const Template: Story = () => {
  const {
    register,
    formState: { errors },
  } = useForm<FormInputs>()
  const validate = {
    email: {
      required: 'メールアドレスを入力してください',
      pattern: {
        value: /[\w\-\\._]+@[\w\-\\._]+\.[A-Za-z]+/,
        message: 'メールアドレスを正しく入力してください',
      },
    },
  }
  const props: InputProps = {
    type: 'email',
    name: 'email',
    placeholder: 'メールアドレス',
    register,
    validate,
    errors,
  }
  return <Input {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
