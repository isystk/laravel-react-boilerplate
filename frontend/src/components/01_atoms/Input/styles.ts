import { css, keyframes } from '@emotion/css'

export const input = css``
