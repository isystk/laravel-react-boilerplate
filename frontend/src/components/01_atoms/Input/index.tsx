import React, { FC } from 'react'
import { ContainerProps } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'

/** InputProps Props */
export type InputProps = {
  small: boolean
  type: 'text' | 'password' | 'email'
  name: string
  placeholder: string
  value
  onChange
  register
  validate
  errors
}
/** Presenter Props */
export type PresenterProps = InputProps

/** Presenter Component */
const InputPresenter: FC<PresenterProps> = ({
  small,
  name,
  value = '',
  onChange,
  register,
  validate,
  errors,
  ...props
}) => (
  <>
    {register ? (
      <>
        <input
          {...props}
          className={`${styles.input} ${
            small ? 'p-2' : 'p-3'
          } w-full bg-gray-200 rounded-md`}
          {...register(name, validate[name])}
        />
        {errors[name] && (
          <span className="pt-4 text-red-500">{errors[name].message}</span>
        )}
      </>
    ) : (
      <input
        {...props}
        className={`${styles.input} ${
          small ? 'p-2' : 'p-3'
        } w-full bg-gray-200 rounded-md`}
        value={value}
        onChange={(e) => onChange(e.target.value)}
      />
    )}
  </>
)

/** Container Component */
const InputContainer: React.FC<ContainerProps<InputProps, PresenterProps>> = ({
  presenter,
  ...props
}) => {
  return presenter({
    ...props,
  })
}

export default connect<InputProps, PresenterProps>(
  'Input',
  InputPresenter,
  InputContainer
)
