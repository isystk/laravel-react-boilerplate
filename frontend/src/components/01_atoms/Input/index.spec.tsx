import React from 'react'
import renderer from 'react-test-renderer'
import Input from './index'
import { useForm } from 'react-hook-form'
import { renderHook } from '@testing-library/react-hooks'

type FormInputs = {
  email: string
}

describe('Input', () => {
  it('Match Snapshot', () => {
    const { result } = renderHook(() => useForm<FormInputs>())

    const {
      register,
      formState: { errors },
    } = result.current

    const validate = {
      email: {
        required: 'メールアドレスを入力してください',
        pattern: {
          value: /[\w\-\\._]+@[\w\-\\._]+\.[A-Za-z]+/,
          message: 'メールアドレスを正しく入力してください',
        },
      },
    }
    const props: InputProps = {
      type: 'email',
      name: 'email',
      placeholder: 'メールアドレス',
      register,
      validate,
      errors,
    }

    const component = renderer.create(<Input {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
