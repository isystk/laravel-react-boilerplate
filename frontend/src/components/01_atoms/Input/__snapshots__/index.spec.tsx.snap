// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Match Snapshot 1`] = `
<input
  className="css-0 p-3 w-full bg-gray-200 rounded-md"
  name="email"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="メールアドレス"
  type="email"
/>
`;
